src/new-client/src/App/LiveRates/Tiles.tsx

```diff
@@ -22,21 +22,24 @@ const PanelItems = styled.div`
   grid-gap: 0.25rem;
 `
 
-const [useFilteredSymbols, filteredSymbols$] = bind(
+const [useFilteredCurrencyPairs, filteredCurrencyPairs$] = bind(
   combineLatest([currencyPairs$, selectedCurrency$]).pipe(
     map(([currencyPairs, selectedCurrency]) => {
-      const allSymbols = Object.keys(currencyPairs)
-      return selectedCurrency === ALL_CURRENCIES
-        ? allSymbols
-        : allSymbols.filter((symbol) =>
-            symbol.includes(selectedCurrency as string),
-          )
+      if (selectedCurrency === ALL_CURRENCIES)
+        return Object.values(currencyPairs)
+      const result = { ...currencyPairs }
+      Object.keys(currencyPairs)
+        .filter((symbol) => symbol.includes(selectedCurrency as string))
+        .forEach((symbol) => {
+          delete result[symbol]
```

Comment:
it makes sense to separate the All_currencies condition and other condition, but why we have a delete here?

Reply:
Ups! I was deleting the wrong ones. Thanks for pointing that out!